pkg/client/introduce/client.go

```diff
@@ -9,15 +9,21 @@ package introduce
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 
 	"github.com/google/uuid"
 
-	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/didexchange"
 	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/introduce"
 	"github.com/hyperledger/aries-framework-go/pkg/storage"
 )
 
+const optionsPrefix = "client_options_"
```

Comment:
what are these options that are being stored? is there a more descriptive name?

Reply:
renamed it to `const proposalOptionsPrefix = "proposal_options_"`, when we create a proposal there are a bunch of options (arguments), we cannot pass it directly to the service. We save options to DB and then we are passing them through `Continue(introduce.ActionDependency)` function.

Reply:
i do not understand these “options” and why they are being stored in this manner. is this the introduce request?

Reply:
When we send a proposal request I want to prepare all the data before the state machine starts. The protocol starts from proposal message, according to that logic, a proposal has a bunch of options (combinations). We do not need this data when we send a proposal but we need it on the next `Continue` iteration. To avoid handling action messages by end-user(client) we saved necessary information to the DB and we provided the method `ProtocolDependency(thID string)` which will supply the service with the data (options are converted to dependency).

```
//  Sends public invitation to the destination
// 	sendProposal(&options{
// 		Invitation: invitation,
// 		Destinations: []*service.Destination{
// 			destination,
// 		},
// 	})
// 	Forwards an invitation from one introducee to another
// 	sendProposal(&options{
// 		Destinations: []*service.Destination{
// 			destination1,
// 			destination2,
// 		},
// 	})
```

Reply:
@sandrask @rolsonquadras ^^^